db/metamapper.php

```diff
@@ -0,0 +1,29 @@
+<?php
+/**
+ * Nextcloud - Notes
+ *
+ * This file is licensed under the Affero General Public License version 3 or
+ * later. See the COPYING file.
+ */
+
+namespace OCA\Notes\Db;
+
+use OCP\IDBConnection;
+use OCP\AppFramework\Db\Mapper;
+
+class MetaMapper extends Mapper {
+
+	public function __construct(IDBConnection $db) {
+		parent::__construct($db, 'notes_meta');
+	}
+
+	public function getAll($userId) {
+		$sql = 'SELECT * FROM `*PREFIX*notes_meta` WHERE user_id=?';
+		return $this->findEntities($sql, [$userId]);
+	}
+
+	public function get($userId, $fileId) {
+		$sql = 'SELECT * FROM `*PREFIX*notes_meta` WHERE user_id=? AND file_id=?';
+		return $this->findEntity($sql, [$userId, $fileId]);
+	}
```

Comment:
It is used to detect whether a note was changed (newly calculated ETag of a note differs from the ETag in the database) or not (newly calculated ETag is equal to the one in the database). In the first case, `last-modified` is updated to `now`. Based on this, the note will be fully transferred to all clients on their next request, even if they use `pruneBefore` to reduce the size of the response.

Please note, that this ETag is not equal to the file based ETag of nextcloud/server since it includes all attributes of a note, e.g. favorite tag (and the ETag in the response header of a list response is a completely different one).